backend/infrahub/graphql/types/event.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from graphene import Field, Interface, ObjectType, String
+from graphene.types.generic import GenericScalar
+
+if TYPE_CHECKING:
+    from graphql import GraphQLResolveInfo
+
+
+class EventNodeInterface(Interface):
+    id = String(required=True)
+    event = String(required=True)
+    branch = String(required=False)
+
+    @classmethod
+    def resolve_type(cls, instance: dict[str, Any], info: GraphQLResolveInfo) -> ObjectType:
+        if "event" in instance:
+            return EVENT_TYPES.get(instance["event"], StandardEvent)
```

Comment:
Here it seems like we're expecting `instance` to be a `dict[str, str]` based on the fact that the keys of `EVENT_TYPES` would be `str`. I'm not completely sure where this gets called from though, it looks like we have it as a generic implementation for our interfaces and that `instance` should be an `ObjectType` within Graphene.

Reply:
this part has always been a bit unclear to me but right now instance is a dict that could be nested. this is why it's defined as `dict[str, Any]`
if `event` is present, we can assume the value will be a string